### PR TITLE
Warmup name and expiry

### PR DESCRIPTION
**Pull request by xsl**
Throws an error dialog if expiry/name detector models are not properly initialized

### CODE REVIEW

File: cardscan-ui/src/main/java/com/getbouncer/cardscan/ui/CardScanActivity.kt [hunk 1]
```diff
@@ -381,6 +385,21 @@ class CardScanActivity :
...
+            Log.e(
+                Config.logTag,
+                "Attempting to run name and expiry without initializing text detector. " +
+                        "Please invoke the warmup() function with initializeNameAndExpiryExtraction to true."
+            )
+            onInvalidApiKey()
+            AlertDialog.Builder(this)
+                .setTitle("Config Problem")
+                .setMessage("Please initialize name/expiry models first in the warmup() function")
+                .setPositiveButton("Ok") { _, _ -> userCancelScan() }
```
Comment (awushensky) [inline cardscan-ui/src/main/java/com/getbouncer/cardscan/ui/CardScanActivity.kt:398]: Please make these localizable by adding them to the `strings.xml` file.

File: cardscan-ui/src/main/java/com/getbouncer/cardscan/ui/CardScanActivity.kt [hunk 2]
```diff
@@ -145,18 +146,20 @@ class CardScanActivity :
...
+
         /**
          * Warm up the analyzers for card scanner. This method is optional, but will increase the
          * speed at which the scan occurs.
          *
          * @param context: A context to use for warming up the analyzers.
          */
         @JvmStatic
-        fun initializeNameExtraction(context: Context, apiKey: String) {
+        fun warmup(context: Context, apiKey: String, initializeNameAndExpiryExtraction: Boolean) {
```
Comment (awushensky) [inline cardscan-ui/src/main/java/com/getbouncer/cardscan/ui/CardScanActivity.kt:158]: can you update the README files to refer to warmup and the new parameter?

File: cardscan-ui/src/main/java/com/getbouncer/cardscan/ui/CardScanActivity.kt [hunk 3]
```diff
@@ -521,6 +516,15 @@ class CardScanActivity :
         scanState = State.FOUND
     }
 
+    private fun showNameAndExpiryInitializationError() {
+        AlertDialog.Builder(this)
+            .setTitle(com.getbouncer.scan.ui.R.string.bouncer_name_and_expiry_initialization_error)
+            .setMessage(com.getbouncer.scan.ui.R.string.bouncer_name_and_expiry_initialization_error_message)
+            .setPositiveButton(com.getbouncer.scan.ui.R.string.bouncer_name_and_expiry_initialization_error_ok) { _, _ -> userCancelScan() }
```
Comment (awushensky) [inline cardscan-ui/src/main/java/com/getbouncer/cardscan/ui/CardScanActivity.kt:523]: nit: you can `import com.getbouncer.scan.ui.R`

File: cardscan-ui/src/main/java/com/getbouncer/cardscan/ui/CardScanActivity.kt (hunk 2 above)
Comment (xsl) [reply cardscan-ui/src/main/java/com/getbouncer/cardscan/ui/CardScanActivity.kt:159]: added em here: https://github.com/getbouncer/cardscan-demo-android/pull/58

**Pull request merged**